src/cata_tiles.cpp

```diff
@@ -179,66 +215,221 @@ void cata_tiles::get_tile_information(std::string config_path, std::string &json
+    int tilecount = 0;
+    for (int y = 0; y < sy; y += tile_height) {
+        for (int x = 0; x < sx; x += tile_width) {
+            source_rect.x = x;
+            source_rect.y = y;
+
+            SDL_Surface *tile_surf = create_tile_surface();
+            if( tile_surf == nullptr ) {
+                continue;
+            }
+
+            if( SDL_BlitSurface( tile_atlas, &source_rect, tile_surf, &dest_rect ) != 0 ) {
+                dbg( D_ERROR ) << "SDL_BlitSurface failed: " << SDL_GetError();
+            }
 
-                SDL_FreeSurface(tile_surf);
+            if (R >= 0 && R <= 255 && G >= 0 && G <= 255 && B >= 0 && B <= 255) {
+                Uint32 key = SDL_MapRGB(tile_surf->format, 0, 0, 0);
+                SDL_SetColorKey(tile_surf, SDL_TRUE, key);
+                SDL_SetSurfaceRLE(tile_surf, true);
+            }
+
+            SDL_Texture *tile_tex = SDL_CreateTextureFromSurface(renderer, tile_surf);
+
+            if( tile_tex == nullptr ) {
+                dbg( D_ERROR) << "failed to create texture: " << SDL_GetError();
+            }
+
+            /** reuse the surface to make alternate color filtered versions */
+            if( SDL_BlitSurface( shadow_tile_atlas, &source_rect, tile_surf, &dest_rect ) != 0 ) {
+                dbg( D_ERROR ) << "SDL_BlitSurface failed: " << SDL_GetError();
+            }
+
+            SDL_Texture *shadow_tile_tex = SDL_CreateTextureFromSurface(renderer, tile_surf);
+            if( shadow_tile_tex == nullptr ) {
+                dbg( D_ERROR) << "failed to create texture: " << SDL_GetError();
+            }
+
+            if( SDL_BlitSurface( nightvision_tile_atlas, &source_rect, tile_surf, &dest_rect ) != 0 ) {
+                dbg( D_ERROR ) << "SDL_BlitSurface failed: " << SDL_GetError();
+            }
+
+            SDL_Texture *night_tile_tex = SDL_CreateTextureFromSurface(renderer, tile_surf);
+            if( night_tile_tex == nullptr ) {
+                dbg( D_ERROR) << "failed to create texture: " << SDL_GetError();
+            }
 
-                if( tile_tex != nullptr ) {
+            if( SDL_BlitSurface( overexposed_tile_atlas, &source_rect, tile_surf, &dest_rect ) != 0 ) {
+                dbg( D_ERROR ) << "SDL_BlitSurface failed: " << SDL_GetError();
+            }
+
+            SDL_Texture *overexposed_tile_tex = SDL_CreateTextureFromSurface(renderer, tile_surf);
+            if( overexposed_tile_tex == nullptr ) {
+                dbg( D_ERROR) << "failed to create texture: " << SDL_GetError();
+            }
+
+            SDL_FreeSurface(tile_surf);
+
+            if( tile_tex != nullptr ) {
                 tile_values.push_back(tile_tex);
                 tilecount++;
-                }
+            }
+            if( shadow_tile_tex != nullptr ) {
+                shadow_tile_values.push_back(shadow_tile_tex);
+                shadow_tilecount++;
```

Comment:
Why this separate counter? It is only incremented when stuff is pushed to `shadow_tile_values`, so why not use `shadow_tile_values.size()` instead? Also the code that resets that counter is a bit inconsistent: the vector is cleared in `clear()`, but the counter is reset in `reinit` (right after calling `clear`). It's technically correct, but it's quite confusing.